src/editconfig.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include "mainview.h"
+
+#include <QComboBox>
+#include <QDialog>
+#include <QLineEdit>
+#include <QListWidget>
+#include <QSpinBox>
+#include <QTextEdit>
+namespace ui
+{
+	// The main edit config dialog class
+	class CEditConfig : public QDialog
+	{
+	public:
+		CEditConfig( CMainView *parent );
+		QListWidget *m_pEditList;
+		QPushButton *m_pApplyButton;
+		QPushButton *m_pCancelButton;
+	};
+
+	// The popup for config items we want to edit.
+	class CEditConfigPopup : public QDialog
+	{
+		bool applyChanges = false;
+
+	public:
+		CEditConfigPopup( CEditConfig *parent );
+		bool shouldApplyChanges() const;
+		QComboBox *m_pTypeComboBox;
```

Comment:
Also, a small thing I just noticed, this stuff should probably be private with accessor funcs for them

Reply:
so...
```c++
QComboBox *getTypeComboBox()
{
    return m_pTypeComboBox;
}
```
???

Reply:
In fact I suppose this doesn't matter actually, so don't need to bother with this.

Reply:
Okay.